src/main.js

```diff
@@ -200,6 +201,13 @@ app.whenReady().then(async () => {
 		createHelpWindow(mainWindow)
 	})
 
+	ipcMain.handle('upgrade:show', () => {
+		const upgradeWindow = createUpgradeWindow()
+
+		mainWindow.destroy()
+		mainWindow = upgradeWindow
+	})
```

Comment:
**issue (blocking):** restoring window on macOS

In MacOS when an app is closed it is still working without open windows. Then, when the app is re-opened from the dock, it restores it by re-creating the main window. To have it work we need to store a function for it. 

```js
createMainWindow = createUpgradeWindow
```